### PR TITLE
image history: fix walking layers

### PR DESCRIPTION
**Pull request by vrothberg**
libimage did not walk thte layers correctly which was probably inherited by old Podman code.  Fix that by vendoring in the corresponding changes in c/common.

Fixes: #20375

#### Does this PR introduce a user-facing change?

```release-note
Fix a bug in `podman history` not displaying the sizes of certain layers.
```

### CODE REVIEW

**Comment by vrothberg** [comment]
Once green we can merge the PR in c/common https://github.com/containers/common/pull/1697 and revendor.

**Comment by vrothberg** [comment]
Now vendoring in c/common@main.  Ready to merge once green.

**Review by flouthoc — APPROVED** [review]
LGTM

**Pull request merged**
